pkg/v8/chartconfig/chartconfig.go

```diff
@@ -76,7 +76,7 @@ func (c *ChartConfig) newTenantK8sClient(ctx context.Context, clusterConfig Clus
 
 func containsChartConfig(list []*v1alpha1.ChartConfig, item *v1alpha1.ChartConfig) bool {
 	for _, l := range list {
-		if reflect.DeepEqual(item, l) {
+		if item.Name == l.Name {
```

Comment:
I think it would be good to check `DeepEqual` of `Name`, `Namespace`, `Annotations`, `Labels` and `Spec`.

Reply:
Checking only for `ObjectMeta.Name` sound fishy, but the only case we should cover here is creation of missing resource, other differences like `ObjectMeta.Labels` or `Spec` will be handled by updates.

As k8s api seems to only use `ObjectMeta.Name` and `ObjectMeta.Namespace` in order to determine if the resource already exists, I do not see a need to check for more.

Also `ObjectMeta.Namespace` and `ObjectMeta.Annotations` are not defined in desired state, which would lead to always have a difference and therefore always try to create the resource.
https://github.com/giantswarm/cluster-operator/blob/5ab18bd64aa480d4fd98c2e603edbf5a95b52f95/pkg/v8/chartconfig/desired.go#L55-L78

Reply:
Ack. So let's please check namespace at least.